alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs

```diff
@@ -395,81 +488,102 @@ runPLCScript proxy (CostModel cost) scriptbytestring units ds =
       name :: String
       name = show ss
    in case P.fromData info of
-        Nothing -> Fails [line]
+        Nothing -> Fails [PlutusFailure line db]
           where
             line =
-              unlines
-                [ "\nThe 3 arg plutus script (" ++ name ++ ") fails.",
-                  show e,
-                  "The data is: " ++ show dat,
-                  "The redeemer is: " ++ show redeemer,
-                  "The third data argument, does not decode to a context\n" ++ show info
-                ]
-        Just info2 -> Fails [line]
+              pack $
+                unlines
+                  [ "\nThe 3 arg plutus script (" ++ name ++ ") fails.",
+                    show e,
+                    "The data is: " ++ show dat,
+                    "The redeemer is: " ++ show redeemer,
+                    "The third data argument, does not decode to a context\n" ++ show info
+                  ]
+            db = B64.encode . serialize' $ PlutusDebug cm eu scriptbytestring ds PlutusV1
+        Just info2 -> Fails [PlutusFailure line db]
           where
             info3 = show (pretty (info2 :: P.ScriptContext))
             line =
-              unlines
-                [ "\nThe 3 arg plutus script (" ++ name ++ ") fails.",
-                  show e,
-                  "The data is: " ++ show dat,
-                  "The redeemer is: " ++ show redeemer,
-                  "The context is:\n" ++ info3
-                ]
-explain_plutus_failure _proxy scriptbytestring e [redeemer, info] =
+              pack $
+                unlines
+                  [ "\nThe 3 arg plutus script (" ++ name ++ ") fails.",
+                    show e,
+                    "The data is: " ++ show dat,
+                    "The redeemer is: " ++ show redeemer,
+                    "The context is:\n" ++ info3
+                  ]
+            db = B64.encode . serialize' $ PlutusDebug cm eu scriptbytestring ds PlutusV1
+explain_plutus_failure _proxy scriptbytestring e ds@[redeemer, info] cm eu =
   -- A two data argument script.
   let ss :: Script era
       ss = PlutusScript scriptbytestring
       name :: String
       name = show ss
    in case P.fromData info of
-        Nothing -> Fails [line]
+        Nothing -> Fails [PlutusFailure line db]
           where
             line =
-              unlines
-                [ "\nThe 2 arg plutus script (" ++ name ++ ") fails.",
-                  show e,
-                  "The redeemer is: " ++ show redeemer,
-                  "The second data argument, does not decode to a context\n" ++ show info
-                ]
-        Just info2 -> Fails [line]
+              pack $
+                unlines
+                  [ "\nThe 2 arg plutus script (" ++ name ++ ") fails.",
+                    show e,
+                    "The redeemer is: " ++ show redeemer,
+                    "The second data argument, does not decode to a context\n" ++ show info
```

Comment:
Is the textual information here now redundant with the debug information?

Reply:
yep, it is. maybe the human readable part should be whittled down, good point, thanks.